### ISSUE TITLE
implement evaluate_model method in FaouziaTabularClassifier

### ISSUE THREAD

**Issue by hamzasherik** [hamzasherik/Faouzia#13]
In this issue, we'll want to implement the evaluate_model abstract method. We'll want to take the output from the train_model method (weights and bias) and run the model on a training set then output the accuracy of that model.

- [ ] implemented evaluate_model method in FaouziaTabularClassifier
- [ ] confirmed implementation accurately captures accuracy on test set for a real dataset
- [ ] wrote unit tests for the evaluate_model method
- [ ] adjusted code to ensure unit test coverage and ensure test cases pass
